drivers/dac/ad5754r/ad5754r.c

```diff
@@ -0,0 +1,772 @@
+{
+	int ret;
+	uint16_t reg_data;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	reg_data |= no_os_field_prep(mask, reg_val);
+
+	return ad5754r_write(dev, instr_addr, reg_data);
+}
+
+/**
+ * @brief Update DAC register value for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param value - The data to be written to the dac channel.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_ch_register(struct ad5754r_dev *dev,
+				   uint8_t chn,
+				   uint16_t value)
+{
+	int ret;
+
+	if (!dev || chn > AD5754R_NUM_CHANNELS)
+		return -EINVAL;
+
+	ret = no_os_gpio_set_value(dev->gpio_ldac, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	ret = ad5754r_write(dev, AD5754R_PREP_INSTR_ADDR(AD5754R_REG_DAC, chn),
+			    value);
+	if (ret)
+		return ret;
+
+	return no_os_gpio_set_value(dev->gpio_ldac, NO_OS_GPIO_HIGH);
+}
+
+/**
+ * @brief Update DAC register value for all channels.
+ * @param dev - The device structure.
+ * @param value - The data to be written to the dac channel registers.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_all_ch_registers(struct ad5754r_dev *dev,
+					uint16_t value)
+{
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = no_os_gpio_set_value(dev->gpio_ldac, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	ret = ad5754r_write(dev,
+			    AD5754R_PREP_INSTR_ADDR(AD5754R_REG_DAC,
+					    AD5754R_DAC_CH_ALL_ADDR),
+			    value);
+	if (ret)
+		return ret;
+
+	return no_os_gpio_set_value(dev->gpio_ldac, NO_OS_GPIO_HIGH);
+}
+
+/**
+ * @brief Read DAC register value for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param value - The data read from dac channel register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read_dac_ch_register(struct ad5754r_dev *dev,
+				 uint8_t chn,
+				 uint16_t *value)
+{
+	if (!value || (chn >= AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	return ad5754r_read(dev,
+			    AD5754R_PREP_INSTR_ADDR(AD5754R_REG_DAC, chn),
+			    (uint16_t *)value);
+}
+
+/**
+ * @brief Convert DAC code to millivolts for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param code - DAC code
+ * @param value - Millivolts output calculated using DAC code.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_dac_code_to_mvolts(struct ad5754r_dev *dev, uint8_t chn,
+			       uint16_t code, uint16_t *value)
+{
+	if (!dev || !value || (chn >= AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	/* Based on encoding, calculate millivolts */
+	if (dev->dac_ch_range[chn] < AD5754R_SPAN_M5V_TO_5V) {
+		/* For unipolar output range, encoding is straight binary */
+		*value = (ad5754r_gain_values[dev->dac_ch_range[chn]]*dev->vref*code) /
```

Comment:
nit: add spaces before and after the operators to improve readability: `*value = (ad5754r_gain_values[dev->dac_ch_range[chn]] * dev->vref * code) / (1 << AD5754R_MAX_RESOLUTION);`